libnestutil/hashtable-common.h

```diff
@@ -40,12 +40,12 @@
 #ifndef UTIL_GTL_HASHTABLE_COMMON_H_
 #define UTIL_GTL_HASHTABLE_COMMON_H_
 
-#include <sparseconfig.h>
 #include <assert.h>
-#include <stdio.h>
-#include <stddef.h> // for size_t
 #include <iosfwd>
+#include <sparseconfig.h>
+#include <stddef.h>  // for size_t
 #include <stdexcept> // For length_error
+#include <stdio.h>
```

Comment:
Is there a specific reason for re-sorting all includes, which is mandated by the addition of the `music_cont_out_proxy`? If there isn't, please revert to the original order.

This is especially important for files not actually provided by the NEST Initiative, like the ones implementing the Google Sparsetable (i.e. `sparsetable.h`, `libc_allocator_with_realloc.h`, `hashtable-common.h`, `sparseconfig.h`, `template_util.h` in `libnestutil`). We want to have these files in the source directory exactly like they come from the upstream source.

For more information on the order of includes, please see our [C++ coding style guide, section _order of includes_](https://nest.github.io/nest-simulator/coding_guidelines_c++#order-of-includes). For files that are original parts of NEST, you can also run our include-checker `extras/includes_checker.py`.